simulation/simple_sensor_simulator/src/vehicle_simulation/ego_entity_simulation.cpp

```diff
@@ -419,26 +440,20 @@ auto EgoEntitySimulation::getCurrentTwist() const -> geometry_msgs::msg::Twist
 auto EgoEntitySimulation::getCurrentPose(const double pitch_angle = 0.) const
   -> geometry_msgs::msg::Pose
 {
-  Eigen::VectorXd relative_position(3);
-  relative_position(0) = vehicle_model_ptr_->getX();
-  relative_position(1) = vehicle_model_ptr_->getY();
-  relative_position(2) = 0.0;
-  relative_position =
-    quaternion_operation::getRotationMatrix(initial_pose_.orientation) * relative_position;
-
-  geometry_msgs::msg::Pose current_pose;
-  current_pose.position.x = initial_pose_.position.x + relative_position(0);
-  current_pose.position.y = initial_pose_.position.y + relative_position(1);
-  current_pose.position.z = initial_pose_.position.z + relative_position(2);
-  current_pose.orientation = [&]() {
-    geometry_msgs::msg::Vector3 rpy;
-    rpy.x = 0;
-    rpy.y = pitch_angle;
-    rpy.z = vehicle_model_ptr_->getYaw();
-    return initial_pose_.orientation * quaternion_operation::convertEulerAngleToQuaternion(rpy);
-  }();
-
-  return current_pose;
+  const auto relative_position =
+    quaternion_operation::getRotationMatrix(initial_pose_.orientation) * world_relative_position_;
```

Comment:
https://eigen.tuxfamily.org/dox/TopicPitfalls.html

> In short: do not use the auto keywords with Eigen's expressions, unless you are 100% sure about what you are doing. In particular, do not use the auto keyword as a replacement for a Matrix<> type. 

- https://github.com/tier4/scenario_simulator_v2/issues/1302

```cpp
const Eigen::Vector3d relative_position =
```

This should fix it.

Reply:
Thank you for finding problem!
I also think it should be fix.

Reply:
Thank you @hakuturu583, it was my pleasure.